bot/cogs/slowmode.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+from discord import TextChannel
+from discord.ext.commands import Cog, Context, group
+
+from bot.bot import Bot
+from bot.constants import Emojis, MODERATION_ROLES
+from bot.converters import DurationDelta
+from bot.decorators import with_role_check
+from bot.utils import time
+
+log = logging.getLogger(__name__)
+
+
+class Slowmode(Cog):
+    """Commands for getting and setting slowmode delays of text channels."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @group(name='slowmode', aliases=['sm'], invoke_without_command=True)
+    async def slowmode_group(self, ctx: Context) -> None:
+        """Get and set the slowmode delay for a given text channel."""
+        await ctx.send_help(ctx.command)
+
+    @slowmode_group.command(name='get', aliases=['g'])
+    async def get_slowmode(self, ctx: Context, channel: TextChannel) -> None:
+        """Get the slowmode delay for a given text channel."""
+        delay = relativedelta(seconds=channel.slowmode_delay)
+
+        # Say "0 seconds" instead of "less than a second"
+        if channel.slowmode_delay == 0:
+            humanized_delay = '0 seconds'
+        else:
+            humanized_delay = time.humanize_delta(delay)
+
+        await ctx.send(f'The slowmode delay for {channel.mention} is {humanized_delay}.')
+
+    @slowmode_group.command(name='set', aliases=['s'])
+    async def set_slowmode(self, ctx: Context, channel: TextChannel, delay: DurationDelta) -> None:
+        """Set the slowmode delay for a given text channel."""
```

Comment:
We definitely should not have to provide the channel for this command. Use a greedy converter to get the channel optionally. If a channel is provided as the first argument, target that channel. Otherwise, use `ctx.channel`.

If I'm in #python-general, I want to be able to do `!slowmode set 3S`, but I also want the option of being able to do `!slowmode set #python-general 3S` from #mods if desired.

Reply:
For an example of a greedy converter, see the `!clean` command.

Reply:
I don't think the greedy converter is necessary here. `typing.Optional` should work fine, since we only want one channel.